spark/src/test/scala/org/apache/spark/sql/delta/perf/OptimizeMetadataOnlyDeltaQuerySuite.scala

```diff
@@ -188,99 +640,261 @@ class OptimizeMetadataOnlyDeltaQuerySuite
+      "LocalRelation [none#0]")
   }
 
-  test("Select Count: group by") {
-    checkSameQueryPlanAndResults(
-      s"SELECT group, COUNT(*) FROM $testTableName GROUP BY group",
-      Seq(Row(1L, totalRows)))
+  test("count-min-max - not supported - group by") {
+    checkOptimizationIsNotTriggered(
+      s"SELECT group, COUNT(*) FROM $testTableName GROUP BY group")
+
+    checkOptimizationIsNotTriggered(
+      s"SELECT group, MIN(id), MAX(id) FROM $testTableName GROUP BY group")
   }
 
-  test("Select Count: count twice") {
-    checkSameQueryPlanAndResults(
-      s"SELECT COUNT(*), COUNT(*) FROM $testTableName",
-      Seq(Row(totalRows, totalRows)))
+  test("count-min-max - not supported - plus literal") {
+    checkOptimizationIsNotTriggered(
+      s"SELECT COUNT(*) + 1 FROM $testTableName")
+
+    checkOptimizationIsNotTriggered(
+      s"SELECT MAX(id) + 1 FROM $testTableName")
   }
 
-  test("Select Count: plus literal") {
-    checkSameQueryPlanAndResults(
-      s"SELECT COUNT(*) + 1 FROM $testTableName",
-      Seq(Row(totalRows + 1)))
+  test("count - not supported - distinct count") {
+    checkOptimizationIsNotTriggered(
+      s"SELECT COUNT(DISTINCT data) FROM $testTableName")
   }
 
-  test("Select Count: distinct") {
-    checkSameQueryPlanAndResults(
-      s"SELECT COUNT(DISTINCT data) FROM $testTableName",
-      Seq(Row(totalDistinctData)))
+  test("count-min-max - not supported - filter") {
+    checkOptimizationIsNotTriggered(
+      s"SELECT COUNT(*) FROM $testTableName WHERE id > 0")
+
+    checkOptimizationIsNotTriggered(
+      s"SELECT MAX(id) FROM $testTableName WHERE id > 0")
   }
 
-  test("Select Count: filter") {
-    checkSameQueryPlanAndResults(
-      s"SELECT COUNT(*) FROM $testTableName WHERE id > 0",
-      Seq(Row(totalRows)))
+  test("count-min-max - not supported - sub-query with filter") {
+    checkOptimizationIsNotTriggered(
+      s"SELECT (SELECT COUNT(*) FROM $testTableName WHERE id > 0)")
+
+    checkOptimizationIsNotTriggered(
+      s"SELECT (SELECT MAX(id) FROM $testTableName WHERE id > 0)")
   }
 
-  test("Select Count: sub-query with filter") {
-    checkSameQueryPlanAndResults(
-      s"SELECT (SELECT COUNT(*) FROM $testTableName WHERE id > 0)",
-      Seq(Row(totalRows)))
+  test("count - not supported - non-null") {
+    checkOptimizationIsNotTriggered(
+      s"SELECT COUNT(ALL data) FROM $testTableName")
+    checkOptimizationIsNotTriggered(
+      s"SELECT COUNT(data) FROM $testTableName")
   }
 
-  test("Select Count: non-null") {
-    checkSameQueryPlanAndResults(
-      s"SELECT COUNT(ALL data) FROM $testTableName",
-      Seq(Row(totalNonNullData)))
-    checkSameQueryPlanAndResults(
-      s"SELECT COUNT(data) FROM $testTableName",
-      Seq(Row(totalNonNullData)))
+  test("count-min-max - not supported - join") {
+    checkOptimizationIsNotTriggered(
+      s"SELECT COUNT(*) FROM $testTableName A, $testTableName B")
```

Comment:
Looks like all these not supported queries can be in a Seq and just create a test for each one of them. Reduces the number of lines in the code and improves readability to see what tests are convered.

Reply:
Done